evidence/pool.go

```diff
@@ -424,25 +263,30 @@ func (evpool *Pool) fastCheck(ev types.Evidence) bool {
 			evpool.logger.Error("Failed to load evidence", "err", err, "evidence", lcae)
 			return false
 		}
-		evInfo, err := bytesToInfo(evBytes)
+		var trustedPb tmproto.LightClientAttackEvidence
+		err = trustedPb.Unmarshal(evBytes)
 		if err != nil {
-			evpool.logger.Error("Failed to convert evidence from proto", "err", err, "evidence", lcae)
+			evpool.logger.Error("Failed to convert evidence from bytes", "err", err, "evidence", lcae)
 			return false
 		}
-		// ensure that all the validators that the evidence pool have found to be malicious
-		// are present in the list of commit signatures in the conflicting block
-	OUTER:
-		for _, sig := range lcae.ConflictingBlock.Commit.Signatures {
-			for _, val := range evInfo.Validators {
-				if bytes.Equal(val.Address, sig.ValidatorAddress) {
-					continue OUTER
-				}
-			}
-			// a validator we know is malicious is not included in the commit
-			evpool.logger.Info("Fast check failed: a validator we know is malicious is not " +
-				"in the commit sigs. Reverting to full verification")
+		trustedEv, err := types.LightClientAttackEvidenceFromProto(&trustedPb)
+		if err != nil {
+			evpool.logger.Error("Failed to convert evidence from protobuf", "err", err, "evidence", lcae)
+			return false
+		}
+		// ensure that all the validators that the evidence pool has matches the validators
+		// in this evidence
+		if len(trustedEv.ByzantineValidators) != len(lcae.ByzantineValidators) {
 			return false
 		}
+		for idx, val := range trustedEv.ByzantineValidators {
+			if !bytes.Equal(lcae.ByzantineValidators[idx].Address, val.Address) {
+				return false
+			}
+			if lcae.ByzantineValidators[idx].VotingPower != val.VotingPower {
```

Comment:
what if they have different `ProposerPriority` or we don't care about that?

Reply:
No I don't think we care about it. Only the address and voting power is sent to the application